hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/javassist/AttributeTypeDescriptor.java

```diff
@@ -79,6 +83,24 @@ public String buildInLineDirtyCheckingBodyFragment(JavassistEnhancementContext c
 		return builder.toString();
 	}
 
+	private void fetchValeBeforeComparison(JavassistEnhancementContext context, CtClass managedCtClass, CtField currentValue, StringBuilder builder) throws NotFoundException {
+		if ( PersistentAttributesHelper.isAssignable(managedCtClass, PersistentAttributeInterceptable.class.getName() )
+				&& context.isLazyLoadable(currentValue)) {
+			// should fetch the value by calling the GET method (if not loaded yet) , in order to initialize the value before the comparison,
+			// otherwise the value = null
+			builder.append( String.format(
+					" if( %1$s != null " +
+							" &&  %1$s instanceof %2$s " +
+							" &&  !((%2$s) %1$s).isAttributeLoaded(\"%3$s\") " +
+							" &&  $1 == null) ",
```

Comment:
I would suggest moving the cheap step to the front to avoid unnecessary performance overhead

Reply:
OK, 
I changed the IF  body to:
" if( $1 == null " +
	" && %1$s != null " +
	" &&  %1$s instanceof %2$s " +
	" &&  !((%2$s) %1$s).isAttributeLoaded(\"%3$s\") ) ",

thanks